streams/src/main/java/org/apache/kafka/streams/processor/internals/ChangelogRecordDeserializationHelper.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.processor.internals;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.header.Header;
+import org.apache.kafka.common.header.internals.RecordHeader;
+import org.apache.kafka.streams.errors.StreamsException;
+import org.apache.kafka.streams.query.Position;
+import org.apache.kafka.streams.state.internals.PositionSerde;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Changelog records without any headers are considered old format.
+ * New format changelog records will have a version in their headers.
+ * Version 0: This indicates that the changelog records have consistency information.
+ */
+public class ChangelogRecordDeserializationHelper {
+    public static final Logger log = LoggerFactory.getLogger(ChangelogRecordDeserializationHelper.class);
+    private static final byte[] V_0_CHANGELOG_VERSION_HEADER_VALUE = {(byte) 0};
+    public static final String CHANGELOG_VERSION_HEADER_KEY = "v";
+    public static final String CHANGELOG_POSITION_HEADER_KEY = "c";
+    public static final RecordHeader CHANGELOG_VERSION_HEADER_RECORD_CONSISTENCY = new RecordHeader(
+            CHANGELOG_VERSION_HEADER_KEY, V_0_CHANGELOG_VERSION_HEADER_VALUE);
+
+    public static Position applyChecksAndUpdatePosition(
+            final ConsumerRecord<byte[], byte[]> record,
+            final boolean consistencyEnabled,
+            final Position position
+    ) {
+        Position restoredPosition = Position.emptyPosition();
+        if (!consistencyEnabled) {
+            return Position.emptyPosition();
```

Comment:
Shouldn't these be returning `position`? This method's contract is to "update" the position, not "replace" it, right?

Reply:
Even though I think this is a bit off, I'm going to go ahead and merge it, so we can fix forward. The overall feature isn't fully implemented yet anyway, so this will have no negative effects if we release the branch right now.

Reply:
It should update the position but `merge` creates a new instance for every invocation. So either we change `merge` to not create a new copy (which is what I prefer since the contract of `merge` is to combine two positions into the first) or I have to manually do the merge which defeats the point of having the `merge` function in the first place

Reply:
Good catch, I totally missed this. Thank you!